packages/edit-site/src/components/sidebar/typography-panel.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	FontSizePicker,
+	__experimentalLineHeightControl as LineHeightControl,
+} from '@wordpress/block-editor';
+import { PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { FONT_SIZE, LINE_HEIGHT, fromPx, toPx } from '../editor/utils';
+
+export default ( {
+	context: { supports, name },
+	getProperty,
+	setProperty,
+} ) => {
+	if (
+		! supports.includes( FONT_SIZE ) &&
+		! supports.includes( LINE_HEIGHT )
+	) {
+		return null;
+	}
+
+	return (
+		<PanelBody title={ __( 'Typography' ) } initialOpen={ true }>
```

Comment:
We are implementing a typography panel two times. One at hooks/block level and one for global styles. I guess the same thing will happen for all the other panels. We probably should think of reusing these panels to avoid a double implementation.
We are using reusing the inner components so I guess this is not a blocker.

Reply:
I tried! Unfortunately, the UI panels we created for hooks are too connected to the implementation they were created for (update block attributes). That gave me a few headaches :sweat_smile: I pondered refactoring to accommodate, but I also want to respect this PR's focus and size (and your time as reviewers). This could be a nice follow-up PR.